openapi-cli/src/main/java/io/ballerina/openapi/generators/schema/ballerinatypegenerators/ReferencedTypeGenerator.java

```diff
@@ -62,6 +62,10 @@ public TypeDescriptorNode generateTypeDescriptorNode() throws BallerinaOpenApiEx
         String typeName = getValidName(extractReferenceType(schema.get$ref()), true);
         Schema<?> refSchema = GeneratorMetaData.getInstance().getOpenAPI().getComponents().getSchemas().get(typeName);
         TypeDescriptorNode typeDescriptorNode = createSimpleNameReferenceNode(createIdentifierToken(typeName));
+        if (refSchema == null) {
+            throw new BallerinaOpenApiException(String.format("'%s' $refs must reference a valid location in" +
+                    " the OpenAPI document.", typeDescriptorNode.toString().trim()));
```

Comment:
Shall we indicate `schema.get$ref()` instead of the `typeDescriptorNode` in the error message?

Reply:
Yeah sure, will change :)